### ISSUE TITLE
UI improvement of default web client

### ISSUE THREAD

**Issue by alci63** [jech/galene#227]
Hi,

I think some small changes would benefit the default UI for occasional users:

1) the logout icon is located just below the close menu icon. More often than not, a user that changed his camera setting will logout instead of closing the menu

2) no feedback on Enable button
When clicking on Enable, no visual feedback is given to the user until the browser asks for permissions, which may take a few seconds. Users often click multiple times, and will quickly conclude it does not work. Teams or Google Meet also ahve this browser "latency", and solve this issue by displaying a message telling user that she will need to give access to her camera / microphone.

3) No button to stop camera
There is a mute button, but no button to stop sharing the camera. It's not clear if Disable will only stop the camera or both camera and microphone.

This are just my 2 (euro) cents, after testing galene with users not knowing anything else than Teams.

**Comment by jech** [comment]
Thanks for the insights.

> 1. the logout icon is located just below the close menu icon.

Good point.  I'm tempted to just remove this icon ­— a user who wants to logout can just close the browser tab.  Do you have any opinions on the subject?

> 2. no feedback on Enable button [...] Users often click multiple times,

The enable button gets disabled once you've clicked it, so even if the user clicks multiple times, no harm will be done.

> no visual feedback is given to the user until the browser asks for permissions, which may take a few seconds

That should not happen, the permissions dialog should appear straight away.  What browser are you using, and what OS?

> 3. There is a mute button, but no button to stop sharing the camera.

Yes, that's by design.  Galene is primarily designed for lecturing, and the idea is that you have a single button to enable both audio and video.

> It's not clear if Disable will only stop the camera or both camera and microphone.

Do you have a suggestions about how to make it clearer?

> after testing galene with users not knowing anything else than Teams

Please be aware that our goal is not to build a clone of *Teams* (or *Zoom*, for that matter), but to build something that's convenient for lectures and presentations.

**Comment by alci63** [comment]
> Thanks for the insights.
> 
> > 1. the logout icon is located just below the close menu icon.
> 
> Good point. I'm tempted to just remove this icon ­— a user who wants to logout can just close the browser tab. Do you have any opinions on the subject?
> 
Yes, removing the button is an option, but I would rather move it to the bottom of the settings menu, to give a clear option and make it explicit to the user that selecting this option will cut any communication and allow to login as another user (with possibly different rights).

> > 2. no feedback on Enable button [...] Users often click multiple times,
> 
> The enable button gets disabled once you've clicked it, so even if the user clicks multiple times, no harm will be done.

If this is the case, it is not very explicit for the user: the button remains clickable, it is deemed on hover, and you clearly can click (maybe it is not handled behind the scene, but the user has no clue of that).
> 
> > no visual feedback is given to the user until the browser asks for permissions, which may take a few seconds
> 
> That should not happen, the permissions dialog should appear straight away. What browser are you using, and what OS?

Firefox on Linux (Kde Plsam / Wayland).

> 
> > 3. There is a mute button, but no button to stop sharing the camera.
> 
> Yes, that's by design. Galene is primarily designed for lecturing, and the idea is that you have a single button to enable both audio and video.
> 
By lecturing, do you mean there is no interaction with the students ? I had the occasion to give lessons and I wanted the students to be able to talk, but not to activate video.

> > It's not clear if Disable will only stop the camera or both camera and microphone.
> 
> Do you have a suggestions about how to make it clearer?
> 
Having two buttons, one for audio and one for video seems a simple choice, isn't it ?

> > after testing galene with users not knowing anything else than Teams
> 
> Please be aware that our goal is not to build a clone of _Teams_ (or _Zoom_, for that matter), but to build something that's convenient for lectures and presentations.

I take note of that point. This is indeed important, but I feel that the difference is not that huge between a lecture and a meeting usage. To me, main difference is probably performance considerations and most attendees not enabling video and audio on big lectures (but probably enabling audio on smaller ones, with say, 30 students, which is what I mostly experimented)

**Comment by alci63** [comment]
Here is a screencast of what happens on my computer

https://github.com/user-attachments/assets/2f615cb9-be0f-4129-8915-ccf2cd6b4cf1

**Comment by jech** [comment]
>>> no visual feedback is given to the user until the browser asks for permissions, which may take a few seconds

>> That should not happen, the permissions dialog should appear straight away. What browser are you using, and what OS?

> Firefox on Linux (Kde Plsam / Wayland).

I've just tried with Firefox, and I cannot reproduce this.  Still, I guess I'll add some feedback.

>>> It's not clear if Disable will only stop the camera or both camera and microphone.

>> Do you have a suggestions about how to make it clearer?

> Having two buttons, one for audio and one for video seems a simple choice, isn't it ?

Having distinct toggles for audio and video is simple but inconvenient.  When you're lecturing, you want to set up your devices once at the beginning of the session, and then switch them on or off as a group.  Suggestions for making that easier to understand are welcome, but the behaviour is not going to change — Zoom's behaviour is quite simply inconvenient.
